Sources/ServiceLifecycle/GracefulShutdown.swift

```diff
@@ -55,6 +55,18 @@ public func withGracefulShutdownHandler<T>(
     return try await operation()
 }
 
+/// Suspend the current task until graceful shutdown is executed.
+///
+/// A common use-case is to call this method in the `operation` closure of ``withGracefulShutdownHandler(operation:onGracefulShutdown:)``
+/// when you are only interested in executing a closure upon graceful shutdown, and nothing else.
```

Comment:
I think we should slightly change this
```suggestion
/// Waits until graceful shutdown is triggered.
///
/// This method suspends the caller until graceful shutdown is triggered. If the calling task is cancelled before
/// graceful shutdown is triggered then this method will exit with a `CancellationError`.
```

Reply:
You think it's not worth mentioning the use-case scenario? I feel like we should mention this somewhere (although maybe the docs for `withGracefulShutdownHandler(operation:onGracefulShutdown:)` are a better place). 
Right now, we don't mention that the graceful shutdown handler won't be executed if the operation finished running, which may or may not be obvious to everyone.